src/main.rs

```diff
@@ -353,7 +362,8 @@ async fn run_loop<T: ApiTrait>(interval: &u32, manager: &CredentialManager<T>, i
                 Ok(_) => info!("Auto sign-up successful."),
                 Err(err) if err == "Invalid credentials" => {
                     error!("Invalid credentials detected.");
-                    show_notification("Your credentials are invalid. Run tuenroll start again");
+                    show_notification("Your credentials are invalid. Run tuenroll start again")
+                        .await;
```

Comment:
What does this `await` do?

Reply:
The function show_notification is async as it checks if the path exists, and downloads the image if necessary (just in case). 
The await is necessary for this to happen.